Library/ViewModels/PledgeViewModel.swift

```diff
@@ -31,43 +40,90 @@ public protocol PledgeViewModelType {
 public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, PledgeViewModelOutputs {
   public init() {
     let projectAndReward = Signal.combineLatest(
-      self.configureProjectAndRewardProperty.signal, self.viewDidLoadProperty.signal
+      self.configureProjectAndRewardProperty.signal,
+      self.viewDidLoadProperty.signal
     )
     .map(first)
     .skipNil()
 
     let project = projectAndReward.map(first)
     let reward = projectAndReward.map(second)
-    let isLoggedIn = projectAndReward
-      .map { _ in AppEnvironment.current.currentUser }
-      .map(isNotNil)
+    let isLoggedIn = projectAndReward.map { _ in AppEnvironment.current.currentUser }.map(isNotNil)
+
+    let pledgeAmount = Signal.merge(
+      self.pledgeAmountSignal,
+      projectAndReward.map { $1.minimum }
+    )
+
+    let shippingRulesEvent = projectAndReward
+      .filter { _, reward in reward.shipping.enabled }
+      .switchMap { (project, reward) -> SignalProducer<Signal<[ShippingRule], ErrorEnvelope>.Event, Never> in
+        AppEnvironment.current.apiService.fetchRewardShippingRules(projectId: project.id, rewardId: reward.id)
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .map(ShippingRulesEnvelope.lens.shippingRules.view)
+          .retry(upTo: 3)
+          .prefix(value: [])
+          .materialize()
+      }
+
+    self.presentShippingRules = shippingRulesEvent.values()
+
+    let selectedShippingRule = Signal.merge(
+      self.viewDidLoadProperty.signal.mapConst(nil),
+      self.presentShippingRules.map(defaultShippingRule(fromShippingRules:))
+    )
 
     let shippingAmount = Signal.merge(
+      selectedShippingRule.skipNil().map { $0.cost },
       self.shippingRuleSignal.map { $0.cost },
       projectAndReward.mapConst(0)
     )
 
-    let pledgeAmount = Signal.merge(
-      self.pledgeAmountSignal,
-      projectAndReward.map { $1.minimum }
+    let shippingShouldBeginLoading = reward
+      .map { $0.shipping.enabled }
+
+    let isShippingLoading = Signal.merge(
+      shippingShouldBeginLoading,
+      shippingRulesEvent.filter { $0.isTerminating }.mapConst(false)
     )
 
-    let total = Signal.combineLatest(pledgeAmount, shippingAmount).map(+)
+    self.shippingRulesError = shippingRulesEvent.errors().map { _ in
+      Strings.We_were_unable_to_load_the_shipping_destinations()
+    }
+
+    let pledgeTotal = Signal.combineLatest(pledgeAmount, shippingAmount).map(+)
 
-    let data = Signal.combineLatest(project, reward, isLoggedIn, total)
-      .map { project, reward, isLoggedIn, total -> PledgeViewData in
-        (project, reward, isLoggedIn, reward.shipping.enabled, total)
+    // swiftlint:disable line_length
+    let data = Signal.combineLatest(project, reward, isLoggedIn, isShippingLoading, selectedShippingRule, pledgeTotal)
```

Comment:
We might be able to improve readability extracting this to a global function in this file?